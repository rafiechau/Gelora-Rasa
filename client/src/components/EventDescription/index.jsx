import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import classes from './style.module.scss';

const EventDescription = ({ description }) => (
  <div className={classes.eventDescription} data-testid="event-description">
    <div className={classes.description}>
      <h2>
        <FormattedMessage id="app_detail_event_description_txt" />
      </h2>
      <div dangerouslySetInnerHTML={{ __html: description }} />
    </div>
  </div>
);

EventDescription.propTypes = {
  description: PropTypes.string,
};

export default EventDescription;
