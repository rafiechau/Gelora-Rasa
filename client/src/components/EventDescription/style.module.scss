$dark-bg: rgba(29, 36, 51, 0.9);
$accent-color: #d32f2f;
$light-color: #fff;
$grey-color: #999;
$spacing-unit: 15px;
$border-radius: 8px;
$fixed-bar-height: 70px;

.eventDescription{
    width: 100%;
    max-width: 1200px;
    padding: $spacing-unit;
    margin-top: $spacing-unit;
    border-radius: $border-radius;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
    color: $light-color;
    .creatorInfo{
        display: flex;
        flex-direction: column;
        margin-bottom: 20px;
        .containerCreator{
            display: flex;
            justify-content: space-between;
            padding: 20px 0;
            button{
                border-color: $accent-color;
                border-radius: 20px;
                padding: 0 20px;
                background-color: $navy;
                color: white;
                cursor: pointer;
            }
        }
        .containerProfil{
            display: flex;
            gap: 2rem;
            align-items: center;
            img{
                border-radius: 100%;
                width: 120px;
                height: 120px;
            }
        }
    }
}