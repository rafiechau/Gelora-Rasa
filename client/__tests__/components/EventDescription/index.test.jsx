import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import EventDescription from '@components/EventDescription';

jest.mock('react-intl', () => ({
  FormattedMessage: jest.fn(({ id }) => <span>{id}</span>),
}));

describe('EventDescription', () => {
  const mockDescription = '<p>This is a test description.</p>';

  test('Should render correctly', () => {
    render(<EventDescription description={mockDescription} />);
    expect(screen.getByTestId('event-description')).toBeInTheDocument();
  });

  test('Should display the correct description', () => {
    render(<EventDescription description={mockDescription} />);
    expect(screen.getByTestId('event-description').innerHTML).toContain(mockDescription);
  });

  test('Should match snapshot', () => {
    const { asFragment } = render(<EventDescription description={mockDescription} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
