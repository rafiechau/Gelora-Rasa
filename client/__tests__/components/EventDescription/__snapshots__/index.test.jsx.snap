// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDescription Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="eventDescription"
    data-testid="event-description"
  >
    <div
      class="description"
    >
      <h2>
        <span>
          app_detail_event_description_txt
        </span>
      </h2>
      <div>
        <p>
          This is a test description.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
